packages/cli/src/databases/repositories/execution.repository.ts

```diff
@@ -66,11 +76,35 @@ function parseFiltersToQueryBuilder(
 
 @Service()
 export class ExecutionRepository extends Repository<ExecutionEntity> {
+	deletionBatchSize = 100;
+
+	hardDeletionInterval: NodeJS.Timer | null = null;
+
 	constructor(
 		dataSource: DataSource,
 		private readonly executionDataRepository: ExecutionDataRepository,
 	) {
 		super(ExecutionEntity, dataSource.manager);
+
+		if (config.getEnv('executions.pruneData')) this.setPruningInterval();
+
+		this.setHardDeletionInterval();
+	}
+
+	setPruningInterval() {
+		setInterval(async () => this.pruneBySoftDeleting(), 1 * TIME.HOUR);
```

Comment:
I think we should add a global AbortController/AbortSignal for the shutdown process, so that we can clean up timers like these properly at shurtdown.

Reply:
Clearing the timers on shutdown at [abe6d9d](https://github.com/n8n-io/n8n/pull/7092/commits/abe6d9df1d298dd9d602fb6198b824e87193beaa), to keep scope small.